### PR TITLE
Editor Messages

### LINKED ISSUE #391: Messages icon in editor
Pretty self descriptive.

### PR DESCRIPTION
**Pull request by TheColaber**
**Resolves**

Resolves #391

**Reason for changes**

I guess it's more convenient...

**Tests**

Works on chrome.

**Notes**

The CSS isn't very good... like it feels weird. If someone has something better, I'll take it.
Any changes that need to be made, let me know.

### CODE REVIEW

File: addons/editor-messages/userscript.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+export default async function ({ addon, global, console }) {
+  while (true) {
+    const nav = await addon.tab.waitForElement("[class^='menu-bar_account-info-group'] [class^='menu-bar_menu-bar-item']", { markAsSeen: true });
+    const messages = document.createElement("a")
+    messages.href = "/messages/"
+    messages.title = "Messages"
+    messages.setAttribute("style", `
```
Comment (WorldLanguages) [inline addons/editor-messages/userscript.js:7]: Why not move all of this CSS to the userstyle?

File: addons/editor-messages/userscript.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    font-size: .85rem;
+    font-weight: bold;
+    `)
+    messages.classList.add("sa-editormessages")
+    let messageCount = document.createElement("span")
+    messages.appendChild(messageCount)
+    if (!nav.querySelector("img")) {
+      nav.appendChild(messages)
+      const setMessages = async () => {
+        fetch(`https://api.scratch.mit.edu/users/${addon.auth.username}/messages/count/`)
```
Comment (WorldLanguages) [inline addons/editor-messages/userscript.js:32]: Use `addon.account.getMsgCount()`

File: addons/editor-messages/userscript.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        .then((response) => response.json())
+        .then((data) => {
+          if (data.count == 0) {
+            messageCount.setAttribute("style", `display: none;`)
+          } else {
+            messageCount.innerText = data.count
+            messageCount.setAttribute("style", `
+            display: block;
```
Comment (WorldLanguages) [inline addons/editor-messages/userscript.js:40]: Move this to the userstyle

File: addons/editor-messages/userscript.js (hunk 2 above)
Comment (TheColaber) [reply addons/editor-messages/userscript.js:32]: It was doing a ton of `console.log`s
Also, anything after that code doesn't run:
```javascript
const setMessages = async () => {
        const msgCount = await addon.account.getMsgCount()
        console.log(msgCount);
      };
setMessages();
setInterval(setMessages, 5000);
```
The console.log here doesn't run...

Comment (WorldLanguages) [reply addons/editor-messages/userscript.js:32]: Haven't we concluded `console.log` works weirdly on editor pages? Can you try something else to test it?

Comment (TheColaber) [reply addons/editor-messages/userscript.js:32]: I tried:
```javascript
const setMessages = async () => {
        // fetch(`https://api.scratch.mit.edu/users/${addon.auth.username}/messages/count/`)
        const msgCount = await addon.account.getMsgCount()
        messageCount.innerText = data.count;
        if (msgCount == 0) {
          messageCount.setAttribute("style", `display: none;`);
        } else {
          messageCount.setAttribute("style", "");
        }
      };
      setMessages();
      setInterval(setMessages, 5000);
```
And the innerText of the messageCount was still left empty...

Comment (WorldLanguages) [reply addons/editor-messages/userscript.js:32]: Why are you printing `data.count`?

Comment (TheColaber) [reply addons/editor-messages/userscript.js:32]: Didn't make a difference...
I can literally add anything after it, and it doesn't produce a single error...

I think everything is wrong with the editor... :(

Comment (WorldLanguages) [reply addons/editor-messages/userscript.js:32]: I mean, I do not understand why you wrote `messageCount.innerText = data.count;`. What's `data.count`?? Did you mean `messageCount`?

Comment (TheColaber) [reply addons/editor-messages/userscript.js:32]: Yeah. that was what I meant, I had it in the code, but it still did not do anything.

**Comment by WorldLanguages** [comment]
@TheColaber I fixed a bug in the addon API where it would never resolve the promise if the message count was 0. Please try using `addon.account.getMsgCount()` again

**Comment by TheColaber** [comment]
Sure!

**Comment by TheColaber** [comment]
Alright, everything is done.

**Review by WorldLanguages — CHANGES_REQUESTED** [review]
image
It looks like this to me. I wonder if it depends on the resolution for one's screen? I've tried fixing it but my CSS skills aren't that good lol

**Comment by TheColaber** [comment]
Yeah, mine either... looks fine on my end but idk how to fix this... someone help me :(

**Comment by WorldLanguages** [comment]
I think `position: absolute;` is not working the same way in the editor navbar vs the website navbar

**Comment by TheColaber** [comment]
@WorldLanguages Does that look better?

**Review by WorldLanguages — CHANGES_REQUESTED** [review]
This solution is still far from ideal, and does not work with message counts with more than one digit. I'll keep this PR on hold until we can have a real CSS solution without any weird hacks.

**Comment by TheColaber** [comment]
> This solution is still far from ideal, and does not work with message counts with more than one digit. I'll keep this PR on hold until we can have a real CSS solution without any weird hacks.

Sounds Good.

**Comment by mxmou** [comment]
@WorldLanguages I improved the CSS a lot, is it good now?

**Comment by TheColaber** [comment]
> @WorldLanguages I improved the CSS a lot, is it good now?

Wow! Looks great for me! After scratch took like 30 minutes to actually load all the APIs, I've finally seen it. It looks awesome.

**Comment by GrahamSH-LLK** [comment]
Scratch is down, that's why.

**Comment by TheColaber** [comment]
> Scratch is down, that's why.

NOOOOOOOOOOOOOOO... _lol an addon that doesn't make scratch be down :(_

**Comment by GrahamSH-LLK** [comment]
And, it's back up!

**Comment by TheColaber** [comment]
YAY! _btw how do you find out these things?_

**Comment by GrahamSH-LLK** [comment]
Uptime robot/

**Comment by WorldLanguages** [comment]
mxmou saving the day with his CSS skills, as always :P

**Comment by TheColaber** [comment]
@mxmou Would you like to credit yourself?

**Comment by mxmou** [comment]
> @mxmou Would you like to credit yourself?

I only made a minor CSS change, so no.

**Comment by TheColaber** [comment]
@WorldLanguages Mind if I also resolve #731 and https://github.com/ScratchAddons/ScratchAddons/commit/20e5fb9d7e53fc69e24467729527bcaf5d11c312#r43982412 in this pr? - they are easy.

**Comment by WorldLanguages** [comment]
@TheColaber No, open separate PRs

**Review by WorldLanguages — APPROVED** [review]

**Pull request merged**